b2bAuthorizer/src/app/modules/metric/metricsUtils.js

```diff
@@ -0,0 +1,64 @@
+function prepareIssuerDimension( decodedJwtToken ) {
+    const issuerId = decodedJwtToken.payload?.iss
+    if(!issuerId) {
+        return {
+            name: "issuer",
+            value: decodedJwtToken.payload?.iss
+        }
+    }
+}
+
+function prepareJWTlifetime( decodedJwtToken ) {
+    return decodedJwtToken.payload?.exp - decodedJwtToken.payload?.iat
+}
+
+function prepareJWTlifepercent( decodedJwtToken ) {
+    return (decodedJwtToken.payload?.exp - new Date()) / (decodedJwtToken.payload?.exp - decodedJwtToken.payload?.iat)
```

Comment:
@ViVeniero io qui metterei ogni valore dentro una variabile, in particolare "new Date()" lo sostituirei con `new Date().getTime() / 1000` 

Poi c'è da capire cosa succede quando `payload` è null, visto che ora usi "?". Io farei un return all'inizio del tipo:
`if(!decodedJwtToken.payload) return 0;` o qualcosa che non faccia pubblicare la metrica.